### ISSUE TITLE
segmentation fault pointer dereference runtime

### ISSUE THREAD

**Issue by iamnobodytoo** [ziglang/zig#2298]
```zig
pub fn main() void {
    var b1: *u8 = &@as(u8, 101);
    b1.* += 1;
}
```

if I do it like this (by using an additional var) it works fine
```
pub fn main() void {
    var ba: u8 = 101;
    var b1: *u8 = &ba; // or &u8(ba)
    b1.* +=1;
}
```

zig version: 0.4.0
system: antergos linux 5.0.7-arch1-1-ARCH

**Comment by andrewrk** [comment]
This should be a compile error, `&u8(101)` should be a `const` pointer.

**Comment by Vexu** [comment]
Duplicate of #2130, fixed by #4971.

```
Semantic Analysis [530/822] ./build/a.zig:270:20: error: expected type '*u8', found '*const u8'
    var b1: *u8 = &@as(u8, 101);
                   ^
./build/a.zig:270:20: note: cast discards const qualifier
    var b1: *u8 = &@as(u8, 101);
                   ^
```
